pkg/kubectl/kubectl.go

```diff
@@ -44,6 +46,32 @@ func makeImageList(spec *api.PodSpec) string {
 	return strings.Join(listOfImages(spec), ",")
 }
 
+func NewThirdPartyResourceMapper(gvs []unversioned.GroupVersion, gvks []unversioned.GroupVersionKind) (meta.RESTMapper, error) {
+	mapper := meta.NewDefaultRESTMapper(gvs, func(gv unversioned.GroupVersion) (*meta.VersionInterfaces, error) {
+		for ix := range gvs {
+			if gvs[ix].Group == gv.Group && gvs[ix].Version == gv.Version {
+				return &meta.VersionInterfaces{
+					Codec:            api.Codec,
+					ObjectConvertor:  api.Scheme,
+					MetadataAccessor: meta.NewAccessor(),
+				}, nil
+			}
+		}
+		g, _ := latest.Group("metrics")
```

Comment:
Use a constant and add a comment explaining why this is hardcoded.  It's not obvious to me.